ceph/ceph_admin/__init__.py

```diff
@@ -138,6 +138,37 @@ def set_tool_repo(self, repo=None):
             for node in self.cluster.get_nodes():
                 node.exec_command(sudo=True, cmd=cmd)
 
+    def set_crimson_repo(self, repo=None):
+        """
+        Add crimson repo on every node part of the cluster.
+
+        Args:
+            repo (Str): repository (default: None)
+
+        """
+        cloud_type = self.config.get("cloud-type", "openstack")
+        logger.info(f"cloud type is {cloud_type}")
+        crimson_repo = self.config.get("crimson_repo", repo)
+        EPEL_REPOS = {
+            "7": "https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm",
```

Comment:
What is the difference b/w these repos and the normal upstream repos that we add?

Since this is mostly upstream, can we make use of the same? and enhance the same for crimson as well?

Reply:
I have removed the additional module which was added for crimson and using the existing setup_upstream_repository method for my workflow